### PR TITLE
Feature ui/dashboard polling frequency

### PR DESCRIPTION
**Pull request by anwar6953**
A couple of styling changes are still needed, but otherwise its good to go.

### CODE REVIEW

File: cdap-ui/app/features/dashboard/templates/widgets/table.html [hunk 1]
```diff
@@ -6,7 +6,12 @@
     </thead>
     <tbody>
     <tr ng-repeat="row in tableData">
-      <td ng-repeat="value in row track by $index">{{::value}}</td>
+      <td ng-repeat="value in row track by $index">
+        <!--dddd, MMMM Do YYYY, h:mm:ss a-->
+        <div ng-if="$first">{{::value * 1000 | amDateFormat:'M/D/YY h:mm:ss a'}}</div>
```
Comment (elfenheart) [inline cdap-ui/app/features/dashboard/templates/widgets/table.html]: I don't think you want to do a single bind in here, it won't update with the frequency

Comment (anwar6953) [inline cdap-ui/app/features/dashboard/templates/widgets/table.html]: It's not a single bind right? It's just read-only, is my understanding.

File: cdap-ui/app/features/dashboard/controllers/widget.js [hunk 2]
```diff
@@ -84,19 +100,41 @@ angular.module(PKG.name+'.feature.dashboard')
...
+      return '1h';
+    }
+
     var zeroFill = function(resolution, result) {
         // interpolating (filling with zeros) the data since backend returns only metrics at specific time periods
         // instead of for the whole range. We have to interpolate the rest with 0s to draw the graph.
+        if (resolution == 'auto') {
+          resolution = resolutionFromAuto(result.startTime, result.endTime);
+        }
         if (resolution == '1h') {
```
Comment (elfenheart) [inline cdap-ui/app/features/dashboard/controllers/widget.js]: Should use Switch statement here

File: cdap-ui/app/features/dashboard/controllers/widget.js [hunk 3]
```diff
@@ -84,19 +100,41 @@ angular.module(PKG.name+'.feature.dashboard')
...
+      } else if (diff <= 36000) {
+        return '1m';
+      }
+      return '1h';
+    }
+
     var zeroFill = function(resolution, result) {
         // interpolating (filling with zeros) the data since backend returns only metrics at specific time periods
         // instead of for the whole range. We have to interpolate the rest with 0s to draw the graph.
+        if (resolution == 'auto') {
```
Comment (elfenheart) [inline cdap-ui/app/features/dashboard/controllers/widget.js]: use '==='

File: cdap-ui/app/features/dashboard/templates/widgets/table.html (hunk 1 above)
Comment (elfenheart) [inline cdap-ui/app/features/dashboard/templates/widgets/table.html]: It's one time binding: https://docs.angularjs.org/guide/expression#one-time-binding

Comment (anwar6953) [inline cdap-ui/app/features/dashboard/templates/widgets/table.html]: ok, got it.

File: cdap-ui/app/features/dashboard/controllers/widget.js [hunk 4]
```diff
@@ -84,21 +100,47 @@ angular.module(PKG.name+'.feature.dashboard')
...
+      switch(resolution) {
+        case '1h':
+          return 60 * 60
+        case '1m':
+            return 60
+        case '1s':
+            return 1
+        default:
+            // backend defaults to '1s'
+            return 1
```
Comment (elfenheart) [inline cdap-ui/app/features/dashboard/controllers/widget.js]: semicolons after return

**Comment by elfenheart** [comment]
:+1: LGTM after semicolons comment

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**